shell/probe/raindrop.ps1

```diff
@@ -1,46 +1,39 @@
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
 function Run {
     Param([String]$Dat = "")
 
-    $TempFile = [System.IO.Path]::GetTempFileName() | Rename-Item -NewName { [System.IO.Path]::ChangeExtension($_, "exe") } -PassThru
-
+    $Vst = New-Item -Path "$Dir\$(New-Guid).exe" -Force
     $Headers = @{
-        'token' = $Token
+        'token' = FromEnv "PRELUDE_TOKEN"
         'dos' = $Dos
         'dat' = $Dat
     }
-    try {
-        $Response = Invoke-WebRequest -URI $Address -UseBasicParsing -Headers $Headers -MaximumRedirection 1 -OutFile $TempFile -PassThru
-    } catch {
-        $StatusCode = $_.Exception.Response.StatusCode.value__
-        Write-Output "ERROR: Failed to reach Prelude Service. " + $StatusCode
-        return
-    }
-    if ($CA -and $CA -ne $Response.BaseResponse.ResponseUri.Authority) {
-        return
-    }
+    $Response = Invoke-WebRequest -URI $Api -UseBasicParsing -Headers $Headers -MaximumRedirection 1 -OutFile $Vst -PassThru
     $Test = $Response.BaseResponse.ResponseUri.AbsolutePath.Split("/")[-1].Split("_")[0]
     if (-not $Test) {
-        Write-Output "INFO: Done running tests"
         return
     }
-
-    $TestExit = Execute $TempFile
-    Start-Process -FilePath $TempFile -ArgumentList "clean" -Wait -NoNewWindow -PassThru
-
-    Remove-Item $TempFile -Force
-    Run -Dat $($Test + ":" + $TestExit)
+    if ($CA -ne $Response.BaseResponse.ResponseUri.Authority) {
+        Write-Output "[P] Bad authority: $Response.BaseResponse.ResponseUri.Authority"
+        exit 0
+    }
+    Write-Output "[P] Running $Test [$Vst]" 
+    $Code = Execute $Vst   
+    Run -Dat "${Test}:$Code"
 }
 
 function Execute { 
     Param([String]$File)
+
     try {
-        return (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru).ExitCode
+        $Code = (Start-Process -FilePath $File -Wait -NoNewWindow -PassThru).ExitCode
+        return If (Test-Path $Vst) {$Code} Else {127}
```

Comment:
question: does it count as PROTECTED if the file was removed after it was already run?

Reply:
also should "$Vst" here be "$File" ?

Reply:
@privateducky first question still stands. Still think that if the file runs, and then its deleted after the run, it's UNPROTECTED, not QUARANTINED

Reply:
we mark it as 127 (quarantined) if the file no longer exists after running it. if something on the system removed the file, I wouldn't mark that as unprotected, I'd mark that as something quarantined our test.

Reply:
if it ran successfully doesn't that mean the bad thing already happened?  If it was quarantined after that then future runs of the test can say PROTECTED, but I'd think that the run that finished would be considered UNPROTECTED